src/libcal_mpi/src/simulation.cpp

```diff
@@ -47,7 +47,7 @@ int cal::mpi_atm_sim::simulate(bool use_cache)
             slice_starts.push_back(ind_start);
             slice_stops.push_back(ind_stop);
 
-            if (slice % ntask == rank) {
+            if (slice % ntask == (size_t) rank) {
```

Comment:
This is a tick to hide a warning! The warning in this point is not related to this specific assegnation but it depends on the nature of the MPI communicator that he wants the rank(s) variable of type `int`, and there is not a way to change it. Basically, I preferred to obtain a warning at this line during the compilation because the variable rank is `int` everywhere.

Reply:
I think there's a way to solve this, but first I need to read the documentation more carefully.